src/components/CourseCard/CourseCard.js

```diff
@@ -1,15 +1,19 @@
-import React from 'react';
+import React, { useState }  from 'react';
 import './CourseCard.scss';
 import { Link } from 'react-router-dom';
 import { addToTeacherCourses } from '../../utils/apiCalls';
 
 const CourseCard = ({ id, category, title, description, thumbnail, badge }) => {
   const route = `courses/${id}`
+  const [error, setError] = useState('');
 
-  const addToMyList = (e) => {
+  const addToMyList = async (e) => {
     e.preventDefault();
-    addToTeacherCourses(id)
-      .then(teacherCourse => console.log(teacherCourse));
+    try {
+      const res = await addToTeacherCourses(id)
+    } catch (error) {
+        setError('Course is already in your lists!')
```

Comment:
Nice Error message here, and nice ui feedback utilization of the space on the card!